gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py

```diff
@@ -3994,8 +3992,8 @@ def impl(context):
     gp_segment_configuration_backup = 'gpexpand.gp_segment_configuration'
 
     query = "select hostname, datadir from gp_segment_configuration where content = -1 order by dbid"
-    conn = dbconn.connect(dbconn.DbURL(dbname='postgres'), unsetSearchPath=False)
-    res = dbconn.execSQL(conn, query).fetchall()
+    with dbconn.connect(dbconn.DbURL(dbname='postgres'), unsetSearchPath=False) as conn:
+        res = dbconn.execSQL(conn, query).fetchall()
```

Comment:
I think the code below should also be moved to the right.

Reply:
Once we have received the rows, we no longer need a connection.

Reply:
compare
```python
class A:
    def __init__(self, a):
        self.a = [a]
    def __enter__(self):
        return self
    def __exit__(self, type, value, traceback):
        self.a = None
    
with A('qwe') as a:
    b = a

print(b.a[0])
```
with
```python
class A:
    def __init__(self, a):
        self.a = [a]
    def __enter__(self):
        return self
    def __exit__(self, type, value, traceback):
        self.a = None
    
with A('qwe') as a:
    b = a

    print(b.a[0])
```